lib/parser.js

```diff
@@ -138,7 +138,12 @@ async function parse(asyncapiYAMLorJSON, options = {}) {
  * @param {Object} [options] Configuration to pass to the {@link module:Parser#parse} method.
  * @returns {Promise<AsyncAPIDocument>} The parsed AsyncAPI document.
  */
-function parseFromUrl(url, fetchOptions = {}, options) {
+function parseFromUrl(url, fetchOptions, options) {
+  //Why not just addinga default to the arguments list?
+  //All function parameters with default values should be declared after the function parameters without default values. Otherwise, it makes it impossible for callers to take advantage of defaults; they must re-specify the defaulted values or pass undefined in order to "get to" the non-default parameters.
+  //To not breat the API by changing argument position and to silet the linter it is just better to move adding
+  if (!fetchOptions) fetchOptions = {};
```

Comment:
`if (fetchOptions === undefined) fetchOptions = {};`

I know that null, or false is also falsey, but I think, that this code should support native defaulting operation in argument (in this only works with `undefined`)

Reply:
why? I saw people approach passing arguments with `null` in case they do not need to pass anything

Reply:
Yes, but only `undefined` works with default value of argument, as you wrote in comment to this line. If it allows false and null, why not just allow undefined?

Reply:
hmmm, not sure I follow, the comment from my side is about removing default value from the argument and handling it inside function. With your approach, if someone passes null, it won't work:

given:
```
function lysy(m) {
  if (!m) m = "lysy jest fajny"
  console.log(m)
}

function lysyUndefined(m) {
  if (m === undefined) m = "lysy jest fajny"
  console.log(m)
}

lysy("lysy jest niefajny")
lysy()
lysy(undefined)
lysy(null)

lysyUndefined("lysy jest niefajny")
lysyUndefined()
lysyUndefined(undefined)
lysyUndefined(null)
```

results in
```
"lysy jest niefajny"

"lysy jest fajny"

"lysy jest fajny"

"lysy jest fajny"

"lysy jest niefajny"

"lysy jest fajny"

"lysy jest fajny"

null
```

I do not want to be `null` 😆

Reply:
I wrote about it :D native defaulting works only with undefined, but if you want support also `null` or `false`, ok :)